src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,62 @@
+import { useContext, useState } from "react";
+import { UserContext } from "../context/UserContext";
+import { Link } from "react-router-dom";
+import { useParams } from "react-router-dom";
+import { postComment } from "../api";
+
+export default function CommentAdder({ setComments }) {
+  const { article_id } = useParams();
+  const [newComment, setNewComment] = useState("");
+  const [feedbackComment, setFeedbackComment] = useState("");
+  const { user } = useContext(UserContext);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    postComment(user, newComment, article_id)
+      .then((newPostedComment) => {
+        setNewComment("");
+        setComments((currComments) => {
+          return [newPostedComment, ...currComments];
+        });
```

Comment:
If this happened before the actual api call it would be optimistic approach and give the user immediate feedback, it does mean you would need to deal with removing it in the catch block.